clients/ctl/admin-ui/src/features/dataset/dataset.slice.ts

```diff
@@ -1,21 +1,21 @@
-import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { createSelector, createSlice, PayloadAction } from "@reduxjs/toolkit";
 import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
 
 import type { AppState } from "~/app/store";
 import { Dataset, GenerateRequestPayload, GenerateResponse } from "~/types/api";
 
+import { EditableType } from "./types";
+
 export interface State {
-  activeDataset: Dataset | null;
+  activeDatasetFidesKey?: string;
   // collections and fields don't have unique IDs, so we have to use their index
```

Comment:
We may want another refactor to change this index approach. The classification's fields [are going to be](https://github.com/ethyca/fidesctl-plus/pull/106/files#r974617237) identified by the field name, so we'll already have one place we identify fields by name. Also, I feel like I've seen a discussion somewhere to the effect of "if a field changes name, it's a new field as far as fides is concerned" but I can't recall the specifics.

Reply:
yah this approach was \*shudder\*. the danger here is that we don't enforce the names to be unique. there's an open ticket for this in fideslang: https://github.com/ethyca/fides/pull/718#issuecomment-1148046497